src/fal/planner/tasks.py

```diff
@@ -0,0 +1,159 @@
+from __future__ import annotations
+
+import argparse
+import sys
+import traceback
+import uuid
+from contextlib import contextmanager
+from dataclasses import dataclass, field
+from functools import cached_property
+from typing import Iterator, Any
+
+from dbt.logger import GLOBAL_LOGGER as logger
+
+from fal.node_graph import FalScript
+from fal.utils import print_run_info
+from faldbt.project import DbtModel, FalDbt, NodeStatus
+
+SUCCESS = 0
+FAILURE = 1
+
+
+class Task:
+    def execute(self, args: argparse.Namespace, fal_dbt: FalDbt) -> int:
+        raise NotImplementedError
+
+
+def _unique_id_to_model_name(unique_id: str) -> str:
+    split_list = unique_id.split(".")
+    # if its a unique id 'model.fal_test.model_with_before_scripts'
+    return split_list[len(split_list) - 1]
+
+
+def _unique_ids_to_model_names(id_list: list[str]) -> list[str]:
+    return list(map(_unique_id_to_model_name, id_list))
+
+
+def _mark_dbt_nodes_status(
+    fal_dbt: FalDbt,
+    status: NodeStatus,
+    dbt_node: str | None = None,
```

Comment:
We support Python 3.7

Reply:
True, true (it would probably work since it is under `from __future__ import annotations` but yea let's do these when we drop 3.7).